internal/s3select/select.go

```diff
@@ -677,6 +677,18 @@ func NewS3Select(r io.Reader) (*S3Select, error) {
 type limitedReadCloser struct {
 	io.LimitedReader
 	io.Closer
+	// for io.Closer
+	once sync.Once
+}
+
+func (l *limitedReadCloser) Close() error {
+	var err error
+	l.once.Do(func() {
+		if l.Closer != nil {
+			err = l.Closer.Close()
+		}
+	})
+	return err
```

Comment:
Isn't the point to return the same error each time Close() is called ? because currently, the first call returns l.Closer.Close() but the second call returns <nil>

Reply:
In fact, I think the kind of processing you are talking about is not needed, because usually such errors are ignored and not handled, but of course it is possible to return the first closed error every time. I chose the latter.